### ISSUE TITLE
Encountering problem  when augmented with mask.

### ISSUE THREAD

**Issue by super233** [albumentations-team/albumentations#868]
## 🐛 Bug

**The augmented mask is not match to the augmented image.**

## To Reproduce

```python
aug = A.Compose([A.OneOf([A.HorizontalFlip(p=0.5), A.VerticalFlip(p=0.5), A.Transpose(p=0.5)], p=1),
      A.RandomRotate90(p=0.5),
      A.GridDistortion(p=0.5)])

augmented  = aug(image=image, mask=label)
image = augmented['image']
label= augmented['mask']
```

## Environment

 - Albumentations version (e.g., 0.1.8): 0.4.6
 - Python version (e.g., 3.7): python 3.7.4
 - OS (e.g., Linux): Ubuntu 18.04.3 LTS
 - How you installed albumentations (`conda`, `pip`, source): conda
 - Any other relevant information:

## Additional context
- original images with mask

![image](https://user-images.githubusercontent.com/35358984/113701830-3b48e700-970b-11eb-8d73-cff6972c37c0.png)

- **wrong augmented result, the mask is not match to the image**

![image](https://user-images.githubusercontent.com/35358984/113701770-2b310780-970b-11eb-9760-5ad9445d925e.png)

**Comment by BloodAxe** [comment]
Indeed. It seems you have pretty old version of the library. Would you mind updating to 0.5+ and re-run your pipeline? I beleive it should be fixed in recent releases.

**Comment by super233** [comment]
> Indeed. It seems you have pretty old version of the library. Would you mind updating to 0.5+ and re-run your pipeline? I beleive it should be fixed in recent releases.

I have updated it to version 0.5.2, but I'm sorry that the problem remained. 😢

**Comment by Dipet** [comment]
I can not reproduce the problem.
Please, provide random seed when you can reproduce the problem.
```python
import cv2 as cv
import albumentations as A
import matplotlib.pyplot as plt
import random
import numpy as np

seed = 42
random.seed(seed)
np.random.seed(seed)

img = cv.imread("D:\\img.png")
mask = cv.imread("D:\\mask.png", cv.IMREAD_GRAYSCALE)
mask = (mask > 128).astype(np.uint8)

aug = A.Compose([
      A.OneOf([
            A.HorizontalFlip(p=0.5),
            A.VerticalFlip(p=0.5),
            A.Transpose(p=0.5)
      ], p=1),
      A.RandomRotate90(p=0.5),
      A.GridDistortion(p=0.5)
])

res = aug(image=img, mask=mask)

plt.subplot(221, title="original")
plt.imshow(img, vmin=0, vmax=255)
plt.subplot(222, title="original")
plt.imshow(mask, vmin=0, vmax=1, cmap="gray")

plt.subplot(223, title="result")
plt.imshow(res["image"], vmin=0, vmax=255)
plt.subplot(224, title="result")
plt.imshow(res["mask"], vmin=0, vmax=1, cmap="gray")

plt.show()
```
![image](https://user-images.githubusercontent.com/7512250/113744225-3422d880-970d-11eb-8b67-68e7e9110ec1.png)

**Comment by super233** [comment]
@Dipet @BloodAxe 

One more thing, because the format of image source file was `.nii`, so I firstly convert it to `np.ndarray`.

You can get the image and mask `np.ndarray` file from: 
- https://drive.google.com/file/d/12cXJqkS9nMAYbaMteKFULZnBC59A_2Fc/view?usp=sharing
- https://drive.google.com/file/d/1kC5My4VJmaMylWoJAXyzKZOdhkC6Apse/view?usp=sharing

And this is the full code, you may modify it to reproduce the problem:
```python
# -*- coding: utf-8 -*
from torch.utils.data import Dataset
from torchvision import transforms
import torch
import json
import numpy as np
import os
import albumentations as A


class CrossMoDADataSet(Dataset):

    def __init__(self, image_dir, json_file, resize_size=None, crop_size=None, is_aug=False):
        self.image_list, self.label_list = self.get_path_list(image_dir, json_file)
        self.preprocess = self.build_preprocess(resize_size, crop_size)
        self.augment = self.build_augment() if is_aug else None
        # self.to_tensor_and_normalize = transforms.Compose([transforms.ToTensor(), transforms.Normalize()])  # TODO: modify normalize params
        self.to_tensor_and_normalize = transforms.Compose([transforms.ToTensor()])

        print('Info: Scan {} images from {}'.format(self.__len__(), json_file))

    def __len__(self):
        return len(self.image_list)

    def __getitem__(self, index):
        image = np.load(self.image_list[index])
        label = np.load(self.label_list[index]) if self.label_list is not None else None

        image = (image / np.max(image) * 255).astype(np.uint8)  # scale the value to [0, 255]
        label = label.astype(np.uint8)

        if self.preprocess is not None:
            augmented = self.preprocess(image=image, mask=label)
            image = augmented['image']
            label = augmented['mask']

        if self.augment is not None:
            augmented = self.augment(image=image, mask=label)
            image = augmented['image']
            label = augmented['mask']

        return self.to_tensor_and_normalize(image), torch.from_numpy(label) if label is not None else None

    def get_path_list(self, image_dir, json_file):
        with open(json_file) as f:
            json_data = json.load(f)

        image_list = [os.path.join(image_dir, i['npy_path']) for i in json_data]
        if 'label_path' in json_data[0]:
            label_list = [os.path.join(image_dir, i['label_path']) for i in json_data]
        else:
            label_list = None

        return image_list, label_list

    def build_augment(self):
        return A.Compose([A.OneOf([A.HorizontalFlip(p=0.5),
                                   A.VerticalFlip(p=0.5),
                                   A.Transpose(p=0.5)], p=1),
                          A.RandomRotate90(p=0.5),
                          A.GridDistortion(p=0.5)])
        # return A.Compose([
        #     # A.HorizontalFlip(p=0.5),
        #     # A.VerticalFlip(p=0.5),
        #     # A.Transpose(p=0.5),
        #     A.RandomRotate90(p=0.5),
        #     # A.GridDistortion(p=0.5)
        # ])
        # return None

    def build_preprocess(self, resize_size, crop_size):
        transform_list = []
        if resize_size is not None:
            transform_list.append(A.Resize(height=resize_size[0], width=resize_size[1], p=1))
        if crop_size is not None:
            transform_list.append(A.CenterCrop(height=crop_size[0], width=crop_size[1], p=1))

        return A.Compose(transform_list) if len(transform_list) > 0 else None
```

**Comment by Dipet** [comment]
I again can not reproduce the problem
```python
from torch.utils.data import Dataset
from torchvision import transforms
import numpy as np
import albumentations as A
import matplotlib.pyplot as plt
import random


class CrossMoDADataSet(Dataset):
    def __init__(self):
        self.augment = self.build_augment()
        self.to_tensor_and_normalize = transforms.Compose([transforms.ToTensor()])

    def __len__(self):
        return 1

    def __getitem__(self, index):
        image = np.load("D:\\crossmoda_22_ceT1_40.npy")
        label = np.load("D:\\crossmoda_22_Label_40.npy")

        image = (image / np.max(image) * 255).astype(np.uint8)  # scale the value to [0, 255]
        label = label.astype(np.uint8)
        image_orig = image.copy()
        label_orig = label.copy()

        if self.augment is not None:
            augmented = self.augment(image=image, mask=label)
            image = augmented['image']
            label = augmented['mask']

        return image, label, image_orig, label_orig

    def build_augment(self):
        return A.Compose([A.OneOf([A.HorizontalFlip(p=0.5),
                                   A.VerticalFlip(p=0.5),
                                   A.Transpose(p=0.5)], p=1),
                          A.RandomRotate90(p=0.5),
                          A.GridDistortion(p=0.5)])


seed = 0
random.seed(seed)
np.random.seed(seed)
a, b, c, d = CrossMoDADataSet()[0]

plt.subplot(221, title="original")
plt.imshow(c, vmin=0, vmax=255, cmap="gray")
plt.subplot(222, title="original")
plt.imshow(d, vmin=0, vmax=1, cmap="gray")

plt.subplot(223, title="result")
plt.imshow(a, vmin=0, vmax=255, cmap="gray")
plt.subplot(224, title="result")
plt.imshow(b, vmin=0, vmax=1, cmap="gray")

plt.show()
```
![image](https://user-images.githubusercontent.com/7512250/113851707-73503880-97a4-11eb-9623-a7a6d1947e6d.png)

**Comment by super233** [comment]
Sorry, the problem was caused by I called `__getitem__()` twice to display the result, it's a stupid problem, thanks for your reply.
